src/_pytest/config/argparsing.py

```diff
@@ -355,6 +357,43 @@ def parse_args(self, args=None, namespace=None):
             getattr(args, FILE_OR_DIR).extend(argv)
         return args
 
+    def _parse_optional(self, arg_string):  # pragma: no cover
+        # This method is entirely copied from `argparse.ArgumentParser`, with a
+        # one-line bugfix for https://bugs.python.org/issue26967 so we can disable
```

Comment:
It would be nice to contribute this back upstream. 😁 

How can we get some kind of check where this blows up loudly when the problem no longer exists upstream?

How about:

```python
    py_ver = sys.version_info[:2]    
    assert py_ver < (3, 9), "check if this is still required in Python {}.{}".format(*py_ver)
```

This will warn us to check when pytest starts to get tested in 3.9. When the time comes, the outcome might need to change this a more standard version guard like we use in other parts of the code:

```python
if sys.version_info[:2] < (3, 9):
    def _parse_optional(self, arg_string):  # pragma: no cover
```

Or bump the version of the `assert` check.

Reply:
I opened python/cpython#14316 yesterday 😄 

There's a fairly good chance that this will also land in 3.8, so I suggest we just hold off on merging until we know, and then do exactly the conditional definition you suggest.